### PR TITLE
Enable profiler and profile-guided builds

### PR DESCRIPTION
**Pull request by MrSerth**
Using PGO (Profile-Guided Optimizations), we can potentially improve the overall performance of Poseidon.
See https://go.dev/doc/pgo

### CODE REVIEW

File: cmd/poseidon/main.go [hunk 1]
```diff
@@ -37,6 +38,33 @@ func shutdownSentry() {
 	}
 }
 
+func initProfiling(options *config.Profiling) (cancel func()) {
```
Comment (mpass99) [inline cmd/poseidon/main.go:41]: I learned recently that it is not always faster to pass the arguments as a pointer, especially for small structs. This is due to the stack / heap management. If it catches your interest, take a look [here](https://medium.com/a-journey-with-go/go-introduction-to-the-escape-analysis-f7610174e890), for example.

**Review by mpass99 — APPROVED** [review]
Cool - always on the pulse of the new Go features :rocket:

File: cmd/poseidon/main.go [hunk 2]
```diff
@@ -142,6 +170,9 @@ func main() {
 	initSentry(&config.Config.Sentry)
```
[the remark this answers is not in the document]
Comment (MrSerth) [reply cmd/poseidon/main.go]: Thanks! I included two options, one for the profiling and one for PGO. Unfortunately, I was no longer able to use the automatic PGO detection (based on a file called default.pgo next to the main.go), but needed to find a manual workaround. This, however, will ensure we always tag values correctly before sending them to Sentry.

File: cmd/poseidon/main.go (hunk 1 above)
Comment (MrSerth) [reply cmd/poseidon/main.go]: Thanks for providing that link and the information; I changed the occurrence here.

**Review by mpass99 — APPROVED** [review]

File: cmd/poseidon/main.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (mpass99) [reply cmd/poseidon/main.go]: I'm happy to learn about this kind of magic! Changing variable values on build-time :mage:

**Pull request merged**